### PR TITLE
NVA commons version 1.40.3

### PR DESCRIPTION
**Pull request by monaullah**
_NP-47133 Search publication channel, too many search results when query contains ampersand_

- Updating commons to get bug fix for ampersands in query parameter values not being escaped
- Implement abstract method from commons `RestRequestHandler`: `validateRequest`
- Converted some classes to records

### CODE REVIEW

File: src/main/java/no/sikt/nva/pubchannels/handler/create/series/CreateSeriesRequest.java [hunk 1]
```diff
@@ -3,46 +3,37 @@
...
-        this.name = name;
-        this.printIssn = printIssn;
-        this.onlineIssn = onlineIssn;
-        this.homepage = homepage;
+    public CreateSeriesRequest {
     }
 
-    public String getName() {
+    @Override
+    public String name() {
```
Comment (anetteOlli) [inline src/main/java/no/sikt/nva/pubchannels/handler/create/series/CreateSeriesRequest.java:21]: Hvorfor trengs disse overridesene?

File: src/main/java/no/sikt/nva/pubchannels/handler/create/publisher/CreatePublisherRequest.java [hunk 2]
```diff
@@ -3,38 +3,30 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-public class CreatePublisherRequest {
+public record CreatePublisherRequest(@JsonProperty(NAME_FIELD) String name,
+                                     @JsonProperty(ISBN_PREFIX_FIELD) String isbnPrefix,
+                                     @JsonProperty(HOMEPAGE_FIELD) String homepage) {
```
Comment (anetteOlli) [inline src/main/java/no/sikt/nva/pubchannels/handler/create/publisher/CreatePublisherRequest.java:8]: Same here, remove the excess and let Jackson parse the record exactly as it is.

File: src/main/java/no/sikt/nva/pubchannels/handler/create/series/CreateSeriesRequest.java [hunk 3]
```diff
@@ -3,46 +3,37 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-public class CreateSeriesRequest {
+public record CreateSeriesRequest(@JsonProperty(NAME_FIELD) String name,
```
Comment (anetteOlli) [inline src/main/java/no/sikt/nva/pubchannels/handler/create/series/CreateSeriesRequest.java:6]: same here

**Review by anetteOlli — CHANGES_REQUESTED** [review]
Where is the test that demonstrates the problem?

File: src/main/java/no/sikt/nva/pubchannels/handler/create/journal/CreateJournalRequest.java [hunk 4]
```diff
@@ -3,47 +3,37 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-class CreateJournalRequest {
+public record CreateJournalRequest(@JsonProperty(NAME_FIELD) String name,
+                                   @JsonProperty(PRINT_ISSN_FIELD) String printIssn,
+                                   @JsonProperty(ONLINE_ISSN_FIELD) String onlineIssn,
+                                   @JsonProperty(HOMEPAGE_FIELD) String homepage) {
+
```
[the remark this answers is not in the document]
Comment (monaullah) [reply src/main/java/no/sikt/nva/pubchannels/handler/create/journal/CreateJournalRequest.java]: Thanks Anette! IntelliJ did the conversion automatically, and I didn't look to hard on the result 😄 See latest commit with changes

**Comment by monaullah** [comment]
> Where is the test that demonstrates and documents the problem, and prevents the fix from getting accidentally removed?

Since the bug was in the dependency (where the bug is tested and fixed), I'm not sure its necessary to test this here? See Np 47133 UriWrapper, fix escaping ampersand (https://github.com/BIBSYSDEV/nva-commons/pull/501)

What do you think?

**Review by anetteOlli — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**